### PR TITLE
Add create_advanced_sequence_commit_step to NI-DCPower

### LINKED ISSUE #1636: create_advanced_sequence_commit_step method is missing from nidcpower API
### Description of issue

Here's the missing API in LabVIEW: https://www.ni.com/documentation/en/ni-dcpower/20.6/ni-dcpower-node-ref/nidcpower-create-adv-seq-commit-step/

### PR DESCRIPTION
**Pull request by alexdubois-ni**
- [x] This contribution adheres to [CONTRIBUTING.md](https://github.com/ni/nimi-python/blob/master/CONTRIBUTING.md).

- [x] I've updated [CHANGELOG.md](https://github.com/ni/nimi-python/blob/master/CHANGELOG.md) if applicable.

- [x] I've added tests applicable for this pull request

### What does this Pull Request accomplish?

Add support for commit step feature by implementing create_advanced_sequence_commit_step

### List issues fixed by this Pull Request below, if any.

- Fix #1636

### What testing has been done?

Modified system tests pass.

### CODE REVIEW

**Comment by luisgomes252** [comment]
You should link the PR to issue #1636 in the description field.

**Comment by sbethur** [comment]
ok to test

File: src/nidcpower/metadata/functions.py [hunk 1]
```diff
@@ -218,7 +250,7 @@
                 'default_value': True,
                 'direction': 'in',
                 'documentation': {
-                    'description': 'Specifies that this current step in the active sequence is active.'
+                    'description': 'Specifies whether the step created with this function is active in the Active advanced sequence.'
```
Comment (sbethur) [inline src/nidcpower/metadata/functions.py:253]: `function` -> `method`

File: src/nidcpower/metadata/functions.py [hunk 2]
```diff
@@ -195,6 +195,38 @@
...
+            {
+                'direction': 'in',
+                'name': 'channelName',
+                'type': 'ViConstString'
+            },
+            {
+                'default_value': True,
+                'direction': 'in',
+                'documentation': {
+                    'description': 'Specifies whether the step created with this function is active in the Active advanced sequence.'
```
Comment (sbethur) [inline src/nidcpower/metadata/functions.py:221]: `function` -> `method`

File: CHANGELOG.md [hunk 3]
```diff
@@ -33,7 +33,8 @@ All notable changes to this project will be documented in this file.
     * #### Removed
 * ### `nidcpower` (NI-DCPower)
     * #### Added
-        * `get_channel_names` - [#1588](https://github.com/ni/nimi-python/issues/1588) 
+        * `get_channel_names` - [#1588](https://github.com/ni/nimi-python/issues/1588)
+        * `create_advanced_sequence_commit_step`
```
Comment (sbethur) [inline CHANGELOG.md:37]:
```suggestion
        * `create_advanced_sequence_commit_step` - [#1636](https://github.com/ni/nimi-python/issues/1636)
```

**Review by sbethur — APPROVED** [review]

**Comment by sbethur** [comment]
@alexdubois-ni, nidcpower system test is failing:

```
================================== FAILURES ===================================
_______________ test_create_and_delete_advanced_sequence[True] ________________

session = nidcpower.Session(resource_name='4162', channels='0', reset=False, options='Simulate=1, DriverSetup=Model:4162; BoardType:PXIe', independent_channels=True)

    @pytest.mark.channels('0')
    def test_create_and_delete_advanced_sequence(session):
        properties_used = ['output_function', 'voltage_level']
        sequence_name = 'my_sequence'
        session.source_mode = nidcpower.SourceMode.SEQUENCE
        session.create_advanced_sequence(sequence_name=sequence_name, property_names=properties_used, set_as_active_sequence=True)
>       session.create_advanced_sequence_commit_step(set_as_active_step=True)

..\..\src\nidcpower\system_tests\test_system_nidcpower.py:327: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
nidcpower\session.py:73: in aux
    return f(*xs, **kws)
nidcpower\session.py:3664: in create_advanced_sequence_commit_step
    errors.handle_error(self, error_code, ignore_warnings=False, is_error_handling=False)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

session = nidcpower.Session(resource_name='4162', channels='0', reset=False, options='Simulate=1, DriverSetup=Model:4162; BoardType:PXIe', independent_channels=True)
code = -1074118619, ignore_warnings = False, is_error_handling = False

    def handle_error(session, code, ignore_warnings, is_error_handling):
        '''handle_error
    
        Helper function for handling errors returned by nidcpower.Library.
        It calls back into the session to get the corresponding error description
        and raises if necessary.
        '''
    
        if _is_success(code) or (_is_warning(code) and ignore_warnings):
            return
    
        if is_error_handling:
            # The caller is in the midst of error handling and an error occurred.
            # Don't try to get the description or we'll start recursing until the stack overflows.
            description = ''
        else:
            description = session._get_error_description(code)
    
        if _is_error(code):
>           raise DriverError(code, description)
E           nidcpower.errors.DriverError: -1074118619: This device does not support the requested operation.  Refer to the device documentation to determine which operations it supports.
E           
E           Device: 4162

nidcpower\errors.py:102: DriverError
_ test_create_and_delete_advanced_sequence_repeated_capabilities[True-Dev1/0] _

session = nidcpower.Session(resource_name='Dev1/0:3,Dev2/0:3', channels='', reset=False, options='Simulate=1, DriverSetup=Model:4162; BoardType:PXIe', independent_channels=True)
channels = 'Dev1/0'

    @pytest.mark.resource_name('Dev1/0:3, Dev2/0:3')
    @pytest.mark.parametrize(
        'channels',
        [
            'Dev1/0',
            'Dev1/0-3',
            'Dev2/0',
            'Dev2/0:3',
            'Dev1/0,Dev2/0',
            'Dev1/0:1,Dev2/2-3',
            'Dev1/3,Dev2/0-3'
        ]
    )
    def test_create_and_delete_advanced_sequence_repeated_capabilities(session, channels):
        channels_session = session.channels[channels]
        properties_used = ['output_function', 'voltage_level']
        sequence_name = 'my_sequence'
        channels_session.source_mode = nidcpower.SourceMode.SEQUENCE
        channels_session.create_advanced_sequence(sequence_name=sequence_name, property_names=properties_used, set_as_active_sequence=True)
>       channels_session.create_advanced_sequence_commit_step(set_as_active_step=True)

..\..\src\nidcpower\system_tests\test_system_nidcpower.py:637: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
```

**Review by marcoskirsch — APPROVED** [review]

**Comment by sbethur** [comment]
test this please

**Comment by sbethur** [comment]
test this please

**Comment by sbethur** [comment]
test this please

**Comment by sbethur** [comment]
test this please

**Comment by sbethur** [comment]
test this please

**Comment by sbethur** [comment]
test this please

**Comment by sbethur** [comment]
System tests are failing on 64-bit agents due to infrastructure issues. We have a successful nidcpower system test run on win32, that should be sufficient validation for this PR.

**Pull request merged**